lib/Model/Importer/DebugInfo/DwarfImporter.cpp

```diff
@@ -576,50 +541,49 @@ class DwarfToModelConverter : public BinaryImporterHelper {
             continue;
           }
 
-          const QualifiedType *MemberType = rc_recur getType(ChildDie);
-
+          model::UpcastableType MemberType = rc_recur makeType(ChildDie);
           if (MemberType == nullptr) {
             reportIgnoredDie(Die,
                              "The type of member " + Twine(Index + 1)
                                + " cannot be resolved");
-            rc_return nullptr;
+            rc_return false;
           }
 
           // Create new field
-          auto &Field = Struct->Fields()[Offset];
+          auto &Field = Struct.Fields()[Offset];
           Field.OriginalName() = getName(ChildDie);
-          Field.Type() = *MemberType;
+          Field.Type() = std::move(MemberType);
 
           ++Index;
         }
       }
 
       if (Index == 0) {
         reportIgnoredDie(Die, "Struct has no fields");
-        rc_return nullptr;
+        rc_return false;
       }
 
     } break;
 
     case llvm::dwarf::DW_TAG_union_type: {
-      auto *Union = cast<model::UnionDefinition>(T);
-      Union->OriginalName() = Name;
+      auto &Union = cast<model::UnionDefinition>(*Definition);
+      Union.OriginalName() = Name;
 
       uint64_t Index = 0;
       for (const DWARFDie &ChildDie : Die.children()) {
         if (ChildDie.getTag() == DW_TAG_member) {
-          const QualifiedType *MemberType = rc_recur getType(ChildDie);
+          model::UpcastableType MemberType = rc_recur makeType(ChildDie);
           if (MemberType == nullptr) {
             reportIgnoredDie(Die,
                              "The type of member " + Twine(Index + 1)
                                + " cannot be resolved");
-            rc_return nullptr;
+            rc_return false;
           }
 
           // Create new field
-          auto &Field = Union->Fields()[Index];
+          auto &Field = Union.Fields()[Index];
```

Comment:
`addField`? If you introduce a new method, you also have to adopt it around.

Reply:
Done.